api/aperture/policy/language/v1/policy.proto

```diff
@@ -969,19 +969,31 @@ message PromQL {
   }]; // @gotags: default:"10s"
 }
 
-// Component that emits a constant value as an output signal
-message Constant {
-  // Outputs for the Constant component.
+// Constant value that can be used instead of a signal in ports. Can be set to be invalid.
+message ConstantValue {
+  bool valid = 1;
```

Comment:
This is making it a bit ugly to use. Maybe InvalidValue can be a separate message?

Reply:
Ok, let's ask @tanveergill 
Right now I created it as stated in [task description](https://github.com/fluxninja/aperture/issues/870)

Reply:
@DariaKunoichi,
Perhaps we can reverse the boolean to `invalid` so that providing only a double `value` still works. Can make it a `oneof` `invalid` or `value`

Reply:
@krdln, @DariaKunoichi,
I am not sure how that would work. Proto Unmarshal would expect a field to be a concrete type. How can the same field be decoded as float or string. Maybe I am missing something?

Reply:
I did not know that it will work. But here is a test from go-playground:
https://go.dev/play/p/6lQDfLOHz4_c

And this is excerpt from docs:
> ParseFloat recognizes the string "NaN", and the (possibly signed) strings "Inf" and "Infinity" as their respective special floating point values. It ignores case when matching.

Reply:
Using a string to represent a floating point in the underlying proto message is very misleading. Discussed this with Harjot, and we feel that a dedicated field is needed to represent special numbers instead of a bool in the current proposal. The problem with representing invalid as a oneof is that setting invalid to false is an ambiguous state since value if undefined. Here is the final proposal:

```
message ConstantValue {
  oneof const {
    string special_value = 1; // @gotags: validate:"oneof=NaN +Inf -Inf"
    double value = 2;
  }
}
```

Reply:
Let's rename `ConstantValue` to `ConstantSignal` to keep a consistent terminology

Reply:
ok, will do